### ISSUE TITLE
Custom tick labels are not possible without 'category'

### ISSUE THREAD

**Issue by hardhub** [c3js/c3#2519]
* **C3 version**: v0.6.8
* **D3 version**: v5.7.0
* **Browser**: any
* **OS**: any

I need to show custom x values.
x: string
y: number

Something like this:

```
g3h4cD - 0.1
1fh934 - 1.2
29f0aE - 3.4
980ab8 - 0.2
24fb0m - 2.1
```

I have made some sample where I tried to set text as X data.  
Unfortunately it does not work with 'indexed' type (failed to render).  
I had to switch to category:

![image](https://user-images.githubusercontent.com/4306334/47976317-e8465200-e0c2-11e8-837f-af1c9a5f025c.png)

But in this case it shows tick in wrong place. But I want to use the same tick position as here:

![image](https://user-images.githubusercontent.com/4306334/47976484-a669db80-e0c3-11e8-8073-9f74428d42af.png)

I can specify X as numbers (just index of each string like g3h4cD, 1fh934 in array).  
But in this case I cannot use values collection to show desired x ticks labels.  
In other words I cannot do the following:

```JS
var chart = c3.generate({
    data: {
        x: 'x',
        columns: [
            ['x', 1, 2, 3, 4, 5, 6, 7, 8, 9],
            ['data1', 30, 200, 100, 400, 150, 250, 50, 100, 250]
        ]
    },
    axis: {
        x: {
            type: 'indexed',
            values: ['cat1', 'cat2', 'cat3', 'cat4', 'cat5', 'cat6', 'cat7', 'cat8', 'cat9']
        }
    }
});
```

So I do not know what exactly is bug or feature request:
1) NaN when I want to put strings in x data.
2) Moved x axis ticks for category type.
3) Ignored values collection for indexed type.

But combination of mentioned behaviors does not allow to show very simply case:
Show key-value pairs, where key is string without impact on ticks position.

**Comment by hardhub** [comment]
I think I have found it. It was not obvious that category resets 'centered' option.

So I have to specify it manually:

```
axis: {
        x: {
            type: 'category',
            categories: ['cat1', 'cat2', 'cat3', 'cat4', 'cat5', 'cat6', 'cat7', 'cat8', 'cat9'],
            tick: {
             centered: true
            }
        }
    }
```

So it is implemented already, but behaviour is not clear.

I will close issue.
